### DISCUSSION TITLE
Removing or toggling visibility of a material from scene

### DISCUSSION THREAD

**Discussion by gauravbiyani033** [sceneview/sceneview#249] [General]
Hi, 
Is there a way to remove a material or toggling a material's visibility so that it does not show up in scene ? The use case I want to solve is that user can toggle visibility of this material using a button click. I am currently using sceneview version 0.7.0. Any help will be appreciated. Thanks

**Comment by grassydragon** [comment]
Hi!
You can hide a `Node` completely instead of changing its material: https://github.com/SceneView/sceneview-android/blob/1e7c4879bf203d0f81cb08a3139b6a606618a9a4/sceneview/src/main/java/io/github/sceneview/node/Node.kt#L294

**Comment by gauravbiyani033** [comment]
Thanks for the reply. But according to my understanding, this will hide the entire model. What if I just want to hide a specific part of a model, like hiding a materialInstance ?

**Comment by ThomasGorisse** [comment]
In v4.0.0-rc.1, node visibility is controlled declaratively via Compose:

```kotlin
var showNode by remember { mutableStateOf(true) }

SceneView {
    if (showNode) {
        ModelNode(modelInstance = model)
    }
}

// Toggle with: showNode = false
```

To make a model transparent without removing it, adjust the material's alpha:
```kotlin
model?.materialInstances?.forEach { material ->
    material.setParameter("baseColorFactor", 1f, 1f, 1f, 0f) // fully transparent
}
```

The Compose approach (conditionally including the node) is cleaner and more efficient for toggling visibility.
